python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -480,6 +484,71 @@ def close(self) -> None:
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
 
 
+@DeveloperAPI
+class BufferedSharedMemoryChannel(ChannelInterface):
```

Comment:
Why a new class instead of extending `Channel`?

Reply:
I think we don't really need to inherit any implementation, so I made a new class

Reply:
Oh I meant just to add directly to `Channel` class, instead of making a new class or subclass.

Reply:
oh I see. I found it a little cleaner to implement it this way. But no strong opinion if you prefer that way. Let me know if you want me to change this!

Reply:
Hmm I see, yeah that makes sense. Initially, I was thinking that having too many different classes gets a bit complicated once we start wanting to wrap or subclass multiple of them. But for now this seems fine, probably we should just refactor `ChannelInterface` at some point...